Tests/L2Tests/L2Tests/tests/PersistentStore_L2Test_onStorageExceeded.cpp

```diff
@@ -0,0 +1,221 @@
+
+    private:
+        /** @brief Mutex */
+        std::mutex m_mutex;
+
+};
+
+/**
+ * @brief Constructor for PersistentStore L2 test class
+ */
+PersistentStore_L2Test::PersistentStore_L2Test()
+        : L2TestMocks()
+{
+        TEST_LOG("*********************PersistentStore Constructor**************\n");
+        uint32_t status = Core::ERROR_GENERAL;
+        Core::JSONRPC::Message message;
+        ServiceMock service;
+
+        /* Activate plugin in constructor */
+        status = ActivateService("org.rdk.PersistentStore");
+        EXPECT_EQ(Core::ERROR_NONE, status);
+
+}
+
+/**
+ * @brief Destructor for PersistentStore L2 test class
+ */
+PersistentStore_L2Test::~PersistentStore_L2Test()
+{
+    TEST_LOG("************PersistentStore Destructor***********\n");
+    uint32_t status = Core::ERROR_GENERAL;
+    status = DeactivateService("org.rdk.PersistentStore");
+    EXPECT_EQ(Core::ERROR_NONE, status);
+}
+
+
+/**
+ * @brief called when onStorageExceeded
+ *
+ * @param[in] message from Persistent Store on the change
+ */
+
+void PersistentStore_L2Test::onStorageExceeded(const JsonObject &message)
+{
+    TEST_LOG("onStorageExceeded triggered ***\n");
+    std::unique_lock<std::mutex> lock(m_mutex);
+    std::string str;
+    message.ToString(str);
+
+    TEST_LOG("onStorageExceeded received: %s\n", str.c_str());
+}
+
+
+/********************************************************
+************Test case Details **************************
+** 1. Get value from PersistentStore
+** 2. Set value
+** 3. On Storage Exceeded event triggered when maxsize is exceeded
+*******************************************************/
+
+TEST_F(PersistentStore_L2Test, PersistentStoreStorageExceeded)
+{
+    JSONRPC::LinkType<Core::JSON::IElement> jsonrpc(PERSISTENTSTORE_CALLSIGN, PERSISTENTL2TEST_CALLSIGN);
+    StrictMock<AsyncHandlerMock_PStore> async_handler;
+    uint32_t status = Core::ERROR_GENERAL;
+    JsonObject params;
+    JsonObject result;
+    std::string message;
+    JsonObject expected_status;
+
+
+    status = InvokeServiceMethod("org.rdk.PersistentStore.1", "getValue", params, result);
+    EXPECT_EQ(Core::ERROR_GENERAL, status);
+    EXPECT_FALSE(result["success"].Boolean());
+
+    // Subscribe to the "onStorageExceeded" event.
+	status = jsonrpc.Subscribe<JsonObject>(JSON_TIMEOUT,
+                                    _T("onStorageExceeded"),
+                                     &AsyncHandlerMock_PStore::onStorageExceeded,
+                                     &async_handler);
+                                           
+    EXPECT_EQ(Core::ERROR_NONE, status);
+
+    message = "";
+    expected_status.FromString(message);
+    EXPECT_CALL(async_handler, onStorageExceeded(MatchRequestStatus(expected_status)))
+        .WillOnce(Invoke(this, &PersistentStore_L2Test::onStorageExceeded));
+    
+    
+    params["namespace"] = "namespace1";
+    params["key"] = "key1";
+    params["value"] = "12345678";
```

Comment:
If the onStorageExceeded() is expected here, then the code needs to wait for the signal. See how it is done in onTemperatureThresholdChanged() in SystemService_L2Test.cpp.